routes/merged/project/location/load_balancer.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get api? do
+      load_balancer_list
+    end
+
+    r.on NAME_OR_UBID do |lb_name, lb_id|
+      filter = if lb_name
+        {Sequel[:load_balancer][:name] => lb_name}
+      else
+        {Sequel[:load_balancer][:id] => UBID.to_uuid(lb_id)}
+      end
+
+      filter[:private_subnet_id] = @project.private_subnets_dataset.where(location: @location).select(Sequel[:private_subnet][:id])
+      lb = LoadBalancer.first(filter)
+
+      unless lb
+        response.status = request.delete? ? 204 : 404
+        request.halt
+      end
+
+      r.post "attach-vm" do
+        Authorization.authorize(current_account.id, "LoadBalancer:edit", lb.id)
+        required_parameters = %w[vm_id]
+        request_body_params = Validation.validate_request_body(json_params, required_parameters)
+        vm = Vm.from_ubid(request_body_params["vm_id"])
+
+        unless vm
+          response.status = 404
+          if api?
+            r.halt
+          else
+            flash["error"] = "VM not found"
+            r.redirect "#{@project.path}#{lb.path}"
+          end
+        end
+
+        Authorization.authorize(current_account.id, "Vm:view", vm.id)
+
+        if vm.load_balancer
+          flash["error"] = "VM is already attached to a load balancer"
+          response.status = 400
+          r.redirect "#{@project.path}#{lb.path}"
+        end
+
+        lb.add_vm(vm)
+
+        if api?
+          Serializers::LoadBalancer.serialize(lb, {detailed: true})
+        else
+          flash["notice"] = "VM is attached"
+          r.redirect "#{@project.path}#{lb.path}"
+        end
+      end
+
+      r.post "detach-vm" do
```

Comment:
There is a lot of overlap with attach-vm, we could consider combining the two blocks, similar to https://github.com/ubicloud/ubicloud/pull/2210/commits/5f4871437120f55878c25eb15602df88828dd6ae

That doesn't need to be done before merging, though.

Reply:
Merged at caf663f3945b4992a1498b44e7e102c872ea42f9